receiver/googlecloudmonitoringreceiver/receiver.go

```diff
@@ -5,51 +5,339 @@ package googlecloudmonitoringreceiver // import "github.com/open-telemetry/opent
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
 
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.uber.org/zap"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver/internal"
 )
 
 type monitoringReceiver struct {
-	config *Config
-	logger *zap.Logger
-	cancel context.CancelFunc
+	config            *Config
+	logger            *zap.Logger
+	client            *monitoring.MetricClient
+	metricsBuilder    *internal.MetricsBuilder
+	mutex             sync.RWMutex
+	metricDescriptors map[string]*metric.MetricDescriptor
 }
 
 func newGoogleCloudMonitoringReceiver(cfg *Config, logger *zap.Logger) *monitoringReceiver {
 	return &monitoringReceiver{
-		config: cfg,
-		logger: logger,
+		config:         cfg,
+		logger:         logger,
+		metricsBuilder: internal.NewMetricsBuilder(logger),
+	}
+}
+
+func (mr *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
+	mr.metricDescriptors = make(map[string]*metric.MetricDescriptor)
```

Comment:
lets do this in newGoogleCloudMonitoringReceiver above, rather than here so we don't overwrite mr.metricDescriptors if start is called again?

Reply:
Moved now